### PR TITLE
Add tests for the details component

### PR DESCRIPTION
**Pull request by 36degrees**
https://trello.com/c/lJzDNEvk/642-automated-tests-for-details-component

### CODE REVIEW

File: src/components/details/template.test.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    const $component = $('.govuk-c-details')
+    expect($component.get(0).tagName).toEqual('details')
+  })
+
+  it('includes a nested summary', () => {
+    const { $ } = render('details', examples.default)
+
+    // Look for the summary element within the details element
+    const $summary = $('.govuk-c-details__summary', '.govuk-c-details')
```
Comment (nickcolley) [inline src/components/details/template.test.js]: Didn't even know you could do this, I've been doing `$('...').find('...')` all this time.

Comment (36degrees) [reply src/components/details/template.test.js]: Yeah, I think it's called 'context' in jQuery-land. I've seen in the past people generally aren't that aware of it – hence the comment. Do you think it's clear enough what's it doing?

Comment (nickcolley) [reply src/components/details/template.test.js]: In this specific case I'd just rely on good ol' CSS selectors `$('.govuk-c-details > .govuk-c-details__summary')` which would hopefully be more widely understood.

Comment (36degrees) [reply src/components/details/template.test.js]: That's a good call, though not sure we'd want to use `>` as that would be testing that the summary is a direct descendant - which I don't think we care about?

Comment (nickcolley) [reply src/components/details/template.test.js]: I stealth removed the direct descendant, but wasn't quick enough 😢

Comment (36degrees) [reply src/components/details/template.test.js]: 😁

**Pull request merged**